### ISSUE TITLE
CVE download data not in gzip format

### ISSUE THREAD

**Issue by zkoppert** [sjvermeu/cvechecker#46]
I had a working installation with a daily job running a pull and scan that recently broke. It seems that there is a format issue with the file. I tried removing the file so it would re-download and that didn't work either. Any suggestions? Is this a bug on the nvdcve side?
```
$ pullcves pull
Downloading nvdcve-2.0-Modified.xml... ok (downloaded)
Converting nvdcve-2.0-Modified.xml to CSV...
gzip: nvdcve-2.0-Modified.xml.gz: not in gzip format
-:1: parser error : Document is empty

^
unable to parse -
ok
Gathering differences with last pull... ok
Loading in nvdcve-2.0-modified.csv differences in cvechecker.
Loading CVE data from /usr/local/var/cvechecker/cache/nvdcve-2.0-modified.delta into database
 0 records processed (0 already in db)...
Downloading versions.dat... ok (not downloaded, same file)
```

**Comment by sdwalker** [comment]
The NVD XML feeds have been retired ([phase 1](https://nvd.nist.gov/General/News/XML-Vulnerability-Feed-Retirement), [phase 2](https://nvd.nist.gov/General/News/XML-Vulnerability-Feed-Retirement-Phase-2), [phase 3](https://nvd.nist.gov/General/News/XML-Vulnerability-Feed-Retirement-Phase-3)) in favor of the [JSON](https://nvd.nist.gov/vuln/data-feeds#JSON_FEED) feeds.

**Comment by sjvermeu** [comment]
That's unfortunate. I'll update the download code to use the JSON files
when I return from vacation. Shouldn't be all that hard (just unfortunate
timings).

Wkr,
  Sven

**Comment by cmuller** [comment]
Hello,

Is there by any chance any news regarding the support of JSON feeds?

Cheers,
Christophe

**Comment by sjvermeu** [comment]
Sorry for the delay, I totally forgot about it :-( I've updated the script to use the JSON feeds now. It also required a small update on the code as the JSON parser I've selected (jq) does not preserve formatting of floating points, and my C code wasn't happy about that.

**Comment by cmuller** [comment]
Hello,
Thanks for the update.
I have tested it by recompiling cvechecker and it seems to analyze json feeds and populate the DBs.
However I have a problem of not finding CVEs that were found before.

For example if I type:

```bash
$ echo 'cpe:/a:sqlite:sqlite:3.6.0:::' > small.txt
$ cvechecker -w small.txt 
Adding CPE entries
 - Added watch for cpe:/a:sqlite:sqlite:3.6.0:::
$ cvechecker -r => nothing
```

It is not finding https://nvd.nist.gov/vuln/detail/CVE-2019-8457 which involves this version of sqlite.

It looks like in nvdcve-1.1-2019.json for example only the 2.3 version of CPEs are listed and not the 2.2 versions anymore..

I see for this CVE:
```json
            "cpe_match": [
              {
                "vulnerable": true,
                "cpe23Uri": "cpe:2.3:a:sqlite:sqlite:*:*:*:*:*:*:*:*",
                "versionStartIncluding": "3.6.0",
                "versionEndIncluding": "3.27.2"
              }
            ]
```

But if I try to use a CPE in 2.3 version, I get an error:

```bash
$ echo 'cpe:2.3:a:sqlite:sqlite:3.6.0:*:*:*:*:*:*:*' > small.txt
$ cvechecker -w small.txt 
Adding CPE entries
 ! An error occurred while interpreting CPE on line 0
```

So.. I guess we are stuck until the tool can add CPEs to watch of version 2.3.

**Comment by cmuller** [comment]
I guess when issues are closed github might not send any more notifications, so I will open a new issue with this problem (which had nothing to do with the xml to json format change actually, sorry for posting it here..).

**Comment by sudhackar** [comment]
@sjvermeu  If this has been fixed completely, can we expect a release for this? Latest alpine docker images still have 3.9 which do not work OOTB.

**Comment by sjvermeu** [comment]
I've released cvechecker v4.0, which contains the fixes that were closed
recently (including the new CPE format and the compression support).
